modules/publicize/publicize-jetpack.php

```diff
@@ -411,6 +411,249 @@ function get_services( $filter = 'all' ) {
+									&&
+									! empty( $service_id_done[ $name ][ $connection['external_id'] ] )
+							)
+						)
+					)
+				);
+
+				// Was this connections (OR, old-format service) already Publicized to.
+				$done = (
+							( 1 == get_post_meta( $post_id, $this->POST_DONE . $unique_id, true ) )
+							||
+							( 1 == get_post_meta( $post_id, $this->POST_DONE . $name, true ) )
+				); // New and old style flags.
+
+				// If this one has already been publicized to, don't let it happen again.
+				$disabled = false;
+				if ( $done ) {
+					$disabled = true;
+				}
+
+				/**
+				 * If this is a global connection and this user doesn't have enough permissions to modify
+				 * those connections, don't let them change it.
+				 */
+				$cmeta           = $this->get_connection_meta( $connection );
+				$hidden_checkbox = false;
+				if ( ! $done && ( 0 == $cmeta['connection_data']['user_id'] && ! current_user_can( $this->GLOBAL_CAP ) ) ) {
+					$disabled = true;
+					/**
+					 * Filters the checkboxes for global connections with non-prilvedged users.
+					 *
+					 * @module publicize
+					 *
+					 * @since 3.7.0
+					 *
+					 * @param bool   $checked Indicates if this connection should be enabled. Default true.
+					 * @param int    $post_id ID of the current post
+					 * @param string $name Name of the connection (Facebook, Twitter, etc)
+					 * @param array  $connection Array of data about the connection.
+					 */
+					$hidden_checkbox = apply_filters( 'publicize_checkbox_global_default', true, $post_id, $name, $connection );
+				}
+
+				// Determine the state of the checkbox (on/off) and allow filtering.
+				$checked = ( ( 1 != $skip ) || $done );
+				/**
+				 * Filter the checkbox state of each Publicize connection appearing in the post editor.
+				 *
+				 * @module publicize
+				 *
+				 * @since 2.0.1
+				 *
+				 * @param bool $checked Should the Publicize checkbox be enabled for a given service.
+				 * @param int $post_id Post ID.
+				 * @param string $name Service name.
+				 * @param array $connection Array of connection details.
+				 */
+				$checked = apply_filters( 'publicize_checkbox_default', $checked, $post_id, $name, $connection );
+
+				// Force the checkbox to be checked if the post was DONE, regardless of what the filter does.
+				if ( $done ) {
+					$checked = true;
+				}
+
+				// This post has been handled, so disable everything.
+				if ( $all_done ) {
+					$disabled = true;
+				}
+
+				$label  = sprintf(
+					_x( '%1$s: %2$s', 'Service: Account connected as', 'jetpack' ),
+					esc_html( $this->get_service_label( $name ) ),
+					esc_html( $this->get_display_name( $name, $connection ) )
+				);
+				$active = ! $skip || $done;
+
+				$connection_list[] = array(
+					'unique_id'       => $unique_id,
+					'name'            => $name,
+					'checked'         => $checked,
+					'disabled'        => $disabled,
+					'active'          => $active,
```

Comment:
I'm not sure we need `active` at all.

`checked` is used in the list of checkboxes when the Publicize UI is open. `active` is used in the comma separated list of connections when the Publicize UI is closed.

Technically, it's possible for `active` to be different than `checked`, but only via the `publicize_checkbox_default` filter. But why would we ever want to display different information in the two lists?

Perhaps `checked`/`default`, `already`/`done`, and `disabled` are a better set of properties?

Reply:
> But why would we ever want to display different information in the two lists?

Go with YAGNI and ditch this property?